### PR TITLE
✨ Implemented select token modal and button

### PR DESCRIPTION
**Pull request by YohanTz**
## Before
Screenshot 2023-11-12 at 10 11 58

## After
https://github.com/Zohal-Starknet/zohal-interface/assets/37301269/502fa319-7d92-4e63-afac-5fc131d0708c

## Checklist
- [ ] ✅ All existing tests pass.
- [ ] 📃 Documentation has been updated (if necessary).
- [ ] 🚀 This PR has been tested locally and is ready for review.

### CODE REVIEW

**Comment by akassaei** [comment]
Besides the Modal, I think the Token Selectors should be aligned
I don't like the fact that the width is defined by the length of the token symbol
Wdyt about aligning them all the time ?

File: src/app/trade/_components/chart-header.tsx [hunk 1]
```diff
@@ -3,8 +3,8 @@ export default function ChartHeader() {
   return (
     <div className="flex h-[4.75rem] items-center pl-4">
       <div className="flex h-full items-center gap-4 border-r border-[#2A2E37] pr-10">
-        <img alt="ethereum logo" className="w-8" src="/tokens/ethereum.png" />
-        <span className="whitespace-nowrap">ETH-USD</span>
+        <img alt="ethereum logo" className="w-8" src="/tokens/wbtc.png" />
```
Comment (akassaei) [inline src/app/trade/_components/chart-header.tsx:6]: Change the alt of the image and maybe get the source from the TOKENS object instead ?

File: src/app/trade/_components/choose-token-button.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { TOKENS, type TokenSymbol } from "@zohal/app/_helpers/tokens";
+import Button from "@zohal/app/_ui/button";
+import { ChevronRight } from "@zohal/app/_ui/icons";
+import { useState } from "react";
+
+import ChooseTokenModal from "./choose-token-modal";
+
+type ChooseTokenButtonProps = {
+  onTokenSymbolChange: (newTokenSymbol: TokenSymbol) => void;
+  tokenSymbol: TokenSymbol;
```
Comment (akassaei) [inline src/app/trade/_components/choose-token-button.tsx:10]: JSDoc

File: src/app/trade/_components/choose-token-button.tsx
Comment (akassaei) [inline src/app/trade/_components/choose-token-button.tsx:1]: Is it normal that you don't ahve the "use client" directive from a component that uses states ?

File: src/app/trade/_components/choose-token-button.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  function closeModal() {
+    setOpen(false);
+  }
+
+  return (
+    <>
+      <button
+        className="flex flex-shrink-0 items-center gap-2 rounded-full border border-[#363636] bg-[#1b1d22] p-[0.3125rem]"
```
Comment (akassaei) [inline src/app/trade/_components/choose-token-button.tsx:31]: NIT : It is prefered to use the flex property instead of just flex shrink or flex grow

Comment (akassaei) [reply src/app/trade/_components/choose-token-button.tsx:31]: + 0.3125rem is equal to 5px I think ? Lets keep multiple of 4 in our spacing no ?

File: src/app/trade/_components/choose-token-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import Divider from "@zohal/app/_ui/divider";
+import Input from "@zohal/app/_ui/input";
+import { useMemo, useState } from "react";
+
+type ChooseTokenModalProps = {
+  onOpenChange: (open: boolean) => void;
+  onTokenSymbolChange: (newTokenSymbol: TokenSymbol) => void;
+  open: boolean;
+  tokenSymbol: TokenSymbol;
+};
```
Comment (akassaei) [inline src/app/trade/_components/choose-token-modal.tsx:12]: JSDoc

File: src/app/trade/_components/choose-token-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+export default function ChooseTokenModal(props: ChooseTokenModalProps) {
+  const { onOpenChange, onTokenSymbolChange, open, tokenSymbol } = props;
+  const [searchQuery, setSearchQuery] = useState("");
+
+  const filteredTokens = useMemo(() => {
+    return Object.entries(TOKENS).flatMap(([tokenSymbol, tokenInfo]) => {
```
Comment (akassaei) [inline src/app/trade/_components/choose-token-modal.tsx:18]: Q : why do we need a flatMap here ? Seems that we are just mapping through tokens no ?

Comment (YohanTz) [reply src/app/trade/_components/choose-token-modal.tsx:18]: Flat map is also acting as a filter here, I return [] at the end of it so the element is deleted from the array

File: src/app/trade/_components/choose-token-button.tsx
Comment (YohanTz) [reply src/app/trade/_components/choose-token-button.tsx:1]: No! Totally forgot, updated 👍🏽

File: src/app/trade/_components/choose-token-button.tsx (hunk 3 above)
Comment (YohanTz) [reply src/app/trade/_components/choose-token-button.tsx:31]: Mmh for the first one, there is no tailwind property to set the flex-shrink to 0 only with the shrink property
I agree for the second one 👍🏽

File: src/app/trade/_components/chart-header.tsx [hunk 6]
```diff
@@ -1,10 +1,16 @@
...
+
 /* eslint-disable @next/next/no-img-element */
 export default function ChartHeader() {
   return (
     <div className="flex h-[4.75rem] items-center pl-4">
       <div className="flex h-full items-center gap-4 border-r border-[#2A2E37] pr-10">
-        <img alt="ethereum logo" className="w-8" src="/tokens/ethereum.png" />
-        <span className="whitespace-nowrap">ETH-USD</span>
+        <img
+          alt={`${TOKENS.ETH.name} icon`}
```
Comment (akassaei) [inline src/app/trade/_components/chart-header.tsx:9]: You use ETH name with WBTC icon and text here

**Review by akassaei — APPROVED** [review]

**Pull request merged**